go-controller/pkg/ovn/endpoints.go

```diff
@@ -0,0 +1,124 @@
+package ovn
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"os/exec"
+	"strings"
+	"unicode"
+
+	kapi "k8s.io/client-go/pkg/api/v1"
+)
+
+func (ovn *OvnController) getLoadBalancer(protocol kapi.Protocol) string {
+	var out []byte
+	if protocol == kapi.ProtocolTCP {
+		out, _ = exec.Command(OVN_NBCTL, "--data=bare", "--no-heading",
+			"--columns=_uuid", "find", "load_balancer",
+			"external_ids:k8s-cluster-lb-tcp=yes").CombinedOutput()
+	} else if protocol == kapi.ProtocolUDP {
+		out, _ = exec.Command(OVN_NBCTL, "--data=bare", "--no-heading",
+			"--columns=_uuid", "find", "load_balancer",
+			"external_ids:k8s-cluster-lb-udp=yes").CombinedOutput()
+	}
+	outStr := strings.TrimFunc(string(out), unicode.IsSpace)
+	return string(outStr)
+}
+
+func (ovn *OvnController) createLoadBalancerVIP(lb string, serviceIP string, port int32, ips []string, targetPort int32) error {
+	glog.V(4).Infof("Creating lb with %s, %s, %d, [%v], %d", lb, serviceIP, port, ips, targetPort)
+
+	// With service_ip:port as a VIP, create an entry in 'load_balancer'
+	// key is of the form "IP:port" (with quotes around)
+	key := fmt.Sprintf("\"%s:%d\"", serviceIP, port)
+
+	if len(ips) == 0 {
```

Comment:
In the python watcher, we had a watcher for services too. So when a service got deleted, we would get a service delete event and that inturn would cause len(ips) to be zero. When I had tested it, I would not get endpoint events when the service itself got deleted. Has that changed now? Do you get endpoint event with zero ips when a service gets deleted (without pods getting deleted)?